src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -132,8 +145,49 @@ public enum UnreferencedTypesHandlingOption
             _intrinsicTypesToTreatEmptyStringAsNull.UnionWith(additionalIntrinsicTypesToTreatEmptyStringAsNull);
         }
 
+        private static string? DownloadLatestContract(string projectFolder)
```

Comment:
Is this the conclusion that we download the contract here?
It is weird that a configuration class is doing something like this.
We could always relying some script to download the contract and passing in the path and if we get null we could assume there is nothing downloaded.